packages/teleterm/src/ui/DocumentTerminal/useDocTerminal.ts

```diff
@@ -47,24 +48,40 @@ export default function useDocumentTerminal(doc: Props['doc']) {
   }
 
   function updateDocumentPid(pty: PtyProcess): void {
-    ctx.serviceDocs.update(doc.uri, { pid: pty.getPid() });
+    const pid = pty.getPid();
+    if (pid) {
+      ctx.serviceDocs.update(doc.uri, { pid });
+    }
   }
 
+  const refreshDocumentCwd = useMemo(() => {
```

Comment:
It would be better, but `useMemo` doesn't have the cleanup function :( 
We can move the whole `useEffect` from `DocumentTerminal` to `useDocTerminal` but I'm not sure if `useDocTerminal` should be aware of the terminal ref?

Reply:
I was referring to just this part where canceling the debouncing can be part of this hook too. 
```ts
useEffect(() => {
    return () => refreshDocumentCwd.cancel();
  }, []);
```

Reply:
I added `useDocumentCwdUpdater` hook to have all things related to this in a single place